### PR TITLE
Feat: highlights design with date and user key

### LINKED ISSUE #38: View the time and date a highlight was made

### PR DESCRIPTION
**Pull request by 0tuedon**
in the pr,  while creating a highlight the timeStamp(created_at)  is created and saved with the highlights as unix.

closes #38

### CODE REVIEW

**Comment by 0tuedon** [comment]
Screenshot 2023-06-27 at 20 02 55
 I don't know if this is the best way to handle this.

File: apps/extension/src/types.ts [hunk 1]
```diff
@@ -51,4 +51,8 @@ export interface IHighlight {
    * This is a JSON serialized representation of the range
    */
   range?: string;
+   /** Created time of the highlight
+   * This is a unix time of when the highlight was created
+   */
+  created_at?: number
```
Comment (okjodom) [inline apps/extension/src/types.ts:57]: nit: use camelCase

File: apps/extension/src/utils/unixConverter.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export function convertUnixTimestampToDate(timestamp: number): string {
+    // Convert Unix timestamp to milliseconds
+    const milliseconds = timestamp * 1000;
+  
+    // Create a new Date object
+    const date = new Date(milliseconds);
+  
+    // Define the month names
+    const monthNames = [
```
Comment (okjodom) [inline apps/extension/src/utils/unixConverter.ts:9]: nit: `months`?

File: apps/extension/src/utils/unixConverter.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+      'January', 'February', 'March', 'April', 'May', 'June',
+      'July', 'August', 'September', 'October', 'November', 'December'
+    ];
+  
+    // Get the components of the date
+    const monthIndex = date.getMonth();
+    const month = monthNames[monthIndex];
```
Comment (okjodom) [inline apps/extension/src/utils/unixConverter.ts:16]: nit: `const month = months[date.getMonth()];`

just terse code :)

File: apps/extension/src/components/Highlights.tsx [hunk 4]
```diff
@@ -42,6 +62,23 @@ const HighlightView = ({ text, id }: HighlightViewProps) => {
...
+      <Flex
+        padding={'12px'}
+        justifyContent={'space-between'}
+        alignItems={'center'}
+      >
+        {/* i don't know if this is the best way to handle it  
+           most times the author is an empty string?
+        */}
+        {(author || nostrId?.pubkey) && (
+          <Avatar name={author || nostrId?.pubkey} width={25} height={25} />
```
Comment (okjodom) [inline apps/extension/src/components/Highlights.tsx:74]: nice!

File: apps/extension/src/components/Highlights.tsx [hunk 5]
```diff
@@ -42,6 +62,23 @@ const HighlightView = ({ text, id }: HighlightViewProps) => {
...
+        )}
+        {created_at && (
+          <Text color={theme.palette.secondaryTint} fontWeight={700}>
+            {convertUnixTimestampToDate(created_at)}
```
Comment (okjodom) [inline apps/extension/src/components/Highlights.tsx:78]: nit: not sure the full MDY-Time format makes for good UX.

As follow up, we could iterate this to represent dates as, `N <years/months/weeks/hours> ago` if that makes for better UX.
cc @aassoiants

**Comment by okjodom** [comment]
@0tuedon, looks good. I left some suggestions and small nits. LMK if you've addressed them or when ready to merge

**Review by Extheoisah — APPROVED** [review]

File: apps/extension/src/contentscript.ts [hunk 6]
```diff
@@ -225,11 +225,11 @@ const tryPublishHighlight = async (
     const event = new NDKEvent(ndk);
     event.content = text;
     event.kind = KIND_HIGHLIGHT;
+    event.created_at = Math.floor(Date.now() / 1000);
```
Comment (okjodom) [inline apps/extension/src/contentscript.ts:228]: NDK publish pipeline will add created_at value

File: apps/extension/src/contentscript.ts [hunk 7]
```diff
@@ -225,11 +225,11 @@ const tryPublishHighlight = async (
...
     event.tags = [
       ['r', window.location.href],
       ['range', range, 'colighter'],
     ];
-
     if (colighter.debug === 1) {
       localStorage.setItem(HIGHLIGHT_STORAGE_KEY, JSON.stringify(event));
```
Comment (okjodom) [inline apps/extension/src/contentscript.ts:234]: append `created_at` field only when saving to local storage

**Review by okjodom — APPROVED** [review]

**Pull request merged**